client/rest-high-level/src/main/java/org/elasticsearch/client/MigrationRequestConverters.java

```diff
@@ -48,6 +49,15 @@ static Request submitMigrateTask(IndexUpgradeRequest indexUpgradeRequest) {
         return prepareMigrateRequest(indexUpgradeRequest, false);
     }
 
+    static Request getDeprecationInfo(DeprecationInfoRequest deprecationInfoRequest) {
+        String endpoint = new RequestConverters.EndpointBuilder()
+            .addCommaSeparatedPathParts(deprecationInfoRequest.getIndices())
+            .addPathPartAsIs("_xpack", "migration", "deprecations")
```

Comment:
In 7.0 this should use `_migration/deprecations` as of https://github.com/elastic/elasticsearch/pull/35976, although in 6.x it will have to use `_xpack`.

Reply:
Do u think its worth me doing that in a new PR after this so i can backport it back, similar to the way we did the other PRs?

Reply:
lol i guess i dont have a choice! 
```
   > Warnings: [[GET /{index}/_xpack/migration/deprecations] is deprecated! Use [GET /{index}/_migration/deprecations] instead.]
```